papyri/signature.py

```diff
@@ -0,0 +1,76 @@
+import inspect
+
+
+class Signature:
+    """A wrapper around inspect utilities."""
+
+    def __init__(self, target_item):
+        """
+        Initialize the class.
+
+        Parameters
+        ----------
+        target_item : callable
+            The target item to be assigned.
+
+        """
+        self.target_item = target_item
```

Comment:
This will make it quite hard to serialize/deserialize the object, as you store a reference to the object we are inspecting, but on the deserialization side you won't be able to do this.

So generally we will want to compute the various properties of this object at creation time.

If we wan to also levrage the serialisation/deserialisation properties of our `Node`s as well ans want a Signature Node, it is better to have a class method like `from_object`, so that we can inherit from_json / to_json for free.